autonomy/chain/tx.py

```diff
@@ -232,3 +234,153 @@ def process(
             receipt=receipt,
             event=event,
         )
+
+    # TODO The methods below (_transact() and transact_and_verify()) have been written in
+    # view of a future refactor of this class to avoid dependency on ContractConfigs class.
+    # These methods haven been added to the class rather than rewrite existing transact()
+    # and build() to avoid breaking changes.
+    def _transact(
+        self,
+        build_tx_contract: Contract,
+        build_tx_contract_address: str,
+        build_tx_contract_method: Callable,
+        build_tx_contract_kwargs: Dict,
+        dry_run: bool = False,
+    ) -> Dict:
+        """Make a transaction and return a receipt"""
+        retries = 0
+        tx_dict = None
+        tx_digest = None
+        already_known = False
+        deadline = datetime.now().timestamp() + self.timeout
+
+        if not (
+            inspect.ismethod(build_tx_contract_method)
+            and (
+                build_tx_contract_method.__self__ is build_tx_contract
+                or build_tx_contract_method.__self__ is build_tx_contract.__class__
+            )
+        ):
+            raise ValueError(
+                f"{build_tx_contract_method!r} must be a method of {build_tx_contract!r}"
+            )
```

Comment:
In fact, the build_tx_contract is useless, because we are already passing the Callable build_tx_contract_method... But I added this for consistency in the method transact_and_verify below. Comments welcome.